sql/src/main/java/io/crate/analyze/OrderBy.java

```diff
@@ -177,6 +179,25 @@ public OrderBy map(Function<? super Symbol, ? extends Symbol> replaceFunction) {
         return new OrderBy(Lists2.map(orderBySymbols, replaceFunction), reverseFlags, nullsFirst);
     }
 
+    @Nullable
+    public OrderBy filter(Predicate<? super Symbol> filter) {
```

Comment:
Personally I'm always a bit confused whether `filter()` methods take a predicate that *excludes* items or that *includes* items. Maybe we can rename `filter` to `include` or so? (the full signature  `filter(Predicate<? super Symbol> include)` should then give all the relevant information)

Reply:
Good point. Given that the existing use-case is an `exclude` we may want to use that name and change the logic accordingly?

Reply:
Sounds also good. Renaming the method to `exclude(Predicate<..> predicate)` would be another option.